hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileInfo.java

```diff
@@ -155,18 +157,62 @@ private StoreFileInfo(final Configuration conf, final FileSystem fs, final FileS
     } else if (isHFile(p) || isMobFile(p) || isMobRefFile(p)) {
       // HFile
       if (fileStatus != null) {
-        this.createdTimestamp = fileStatus.getModificationTime();
-        this.size = fileStatus.getLen();
+        this.localStatus = fileStatus;
       } else {
-        FileStatus fStatus = fs.getFileStatus(initialPath);
-        this.createdTimestamp = fStatus.getModificationTime();
-        this.size = fStatus.getLen();
+        this.localStatus = loadAndCacheFileStatus(fs);
       }
+      this.createdTimestamp = localStatus.getModificationTime();
+      this.size = localStatus.getLen();
       this.reference = null;
       this.link = null;
     } else {
       throw new IOException("path=" + p + " doesn't look like a valid StoreFile");
     }
+    if(this.localStatus == null){
+      this.localStatus = loadAndCacheFileStatus(fs);
+    }
+  }
+
+  private FileStatus loadAndCacheFileStatus(final FileSystem fs) throws IOException {
+    FileStatus status = null;
+    if (this.reference != null) {
+      if (this.link != null) {
+        for (int i = 0; i < this.link.getLocations().length; i++) {
+          // HFileLink Reference
+          try {
+            status = link.getFileStatus(fs);
+          } catch (FileNotFoundException ex) {
+            // try the other location
+            throw ex;
```

Comment:
just throw ex it will have no chance to try the other location, see the old code, it should keep the behave with old code.